### PR TITLE
linuxPackages.drbd: 9.2.8 -> 9.2.9

### PR DESCRIPTION
**Pull request by birkb**
## Description of changes

- 6.8 kernel support added
- aligned to Nix standard build environment
- test added
- redundant installPhase and postFixup definitions removed

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Thesola10 — APPROVED** [review]

File: pkgs/os-specific/linux/drbd/driver.nix [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-  name = "drbd-${version}-${kernel.version}";
-  version = "9.2.8";
+stdenv.mkDerivation (finalAttrs: {
+  pname = "drbd";
+  version = "9.2.9";
 
   src = fetchurl {
-    url = "https://pkg.linbit.com//downloads/drbd/9/drbd-${version}.tar.gz";
-    hash = "sha256-LqK1lPucab7wKvcB4VKGdvBIq+K9XtuO2m0DP5XtK3M=";
+    url = "https://pkg.linbit.com//downloads/drbd/9/${finalAttrs.pname}-${finalAttrs.version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/drbd/driver.nix:8]:
```suggestion
    url = "https://pkg.linbit.com//downloads/drbd/9/drbd-${finalAttrs.version}.tar.gz";
```
finalAttrs.pname is in many usecases wrong, as when we would append something like headless to pname, then the download URL would be broken

File: nixos/tests/drbd-driver.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import ./make-test-python.nix ({ lib, pkgs, ... }: {
+  name = "drbd-driver";
+  meta.maintainers = with pkgs.lib.maintainers; [ birkb ];
```
Comment (SuperSandro2000) [inline nixos/tests/drbd-driver.nix:3]:
```suggestion
  meta.maintainers = with lib.maintainers; [ birkb ];
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
as mentioned earlier - we should never use `finalAttrs.pname` in a url

**Review by JohnRTitor — CHANGES_REQUESTED** [review]
Perhaps we could fetch from the GitHub repo instead?

```patch
diff --git a/pkgs/os-specific/linux/drbd/driver.nix b/pkgs/os-specific/linux/drbd/driver.nix
index c2e222ad2f1f..8f4b84800f72 100644
--- a/pkgs/os-specific/linux/drbd/driver.nix
+++ b/pkgs/os-specific/linux/drbd/driver.nix
@@ -1,12 +1,14 @@
-{ stdenv, lib, fetchurl, kernel, nixosTests, flex, coccinelle, python3 }:
+{ stdenv, lib, fetchFromGitHub, kernel, nixosTests, flex, coccinelle, python3 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "drbd";
-  version = "9.2.9";
+  version = "9.2.10";
 
-  src = fetchurl {
-    url = "https://pkg.linbit.com//downloads/drbd/9/${finalAttrs.pname}-${finalAttrs.version}.tar.gz";
-    hash = "sha256-MLG0hGvAjjkNxhszbpnb8fkzWUQC1zPhkeOaU0V5x3g=";
+  src = fetchFromGitHub {
+    owner = "LINBIT";
+    repo = "drbd";
+    rev = "drbd-${finalAttrs.version}";
+    hash = "sha256-O77nVnbk0rSS2vI/+jG1jgDIiFmttGFNUgnw3x6WqPk=";
   };
 
   hardeningDisable = [ "pic" ];
```

**Comment by birkb** [comment]
> Perhaps we could fetch from the GitHub repo instead?
> 
> ```diff
> diff --git a/pkgs/os-specific/linux/drbd/driver.nix b/pkgs/os-specific/linux/drbd/driver.nix
> index c2e222ad2f1f..8f4b84800f72 100644
> --- a/pkgs/os-specific/linux/drbd/driver.nix
> +++ b/pkgs/os-specific/linux/drbd/driver.nix
> @@ -1,12 +1,14 @@
> -{ stdenv, lib, fetchurl, kernel, nixosTests, flex, coccinelle, python3 }:
> +{ stdenv, lib, fetchFromGitHub, kernel, nixosTests, flex, coccinelle, python3 }:
>  
>  stdenv.mkDerivation (finalAttrs: {
>    pname = "drbd";
> -  version = "9.2.9";
> +  version = "9.2.10";
>  
> -  src = fetchurl {
> -    url = "https://pkg.linbit.com//downloads/drbd/9/${finalAttrs.pname}-${finalAttrs.version}.tar.gz";
> -    hash = "sha256-MLG0hGvAjjkNxhszbpnb8fkzWUQC1zPhkeOaU0V5x3g=";
> +  src = fetchFromGitHub {
> +    owner = "LINBIT";
> +    repo = "drbd";
> +    rev = "drbd-${finalAttrs.version}";
> +    hash = "sha256-O77nVnbk0rSS2vI/+jG1jgDIiFmttGFNUgnw3x6WqPk=";
>    };
>  
>    hardeningDisable = [ "pic" ];
> ```

`pkg.linbit.com` contains the cleaned up version of the code that can directly be used. It is the official LinBIT source.

File: pkgs/os-specific/linux/drbd/driver.nix (hunk 1 above)
Comment (birkb) [reply pkgs/os-specific/linux/drbd/driver.nix:8]: Thanks for the hint. What would be the correct variable and do you maybe have a link to the related documentation?

Comment (eclairevoyant) [reply pkgs/os-specific/linux/drbd/driver.nix:8]: Don't use pname, just use the string literal.

**Comment by SuperSandro2000** [comment]
closed in favor of the PR that updates straight to 9.2.12

**Pull request closed without merging**